### PR TITLE
KT-7922 Added Boolean Companion Object

### PR DESCRIPTION
**Pull request by Kerooker**
https://youtrack.jetbrains.com/issue/KT-7922

### CODE REVIEW

**Comment by Kerooker** [comment]
@ilya-g

File: core/metadata.jvm/src/org/jetbrains/kotlin/metadata/jvm/deserialization/ClassMapperLite.kt [hunk 1]
```diff
@@ -63,7 +63,7 @@ object ClassMapperLite {
             add("reflect/KFunction$i", "kotlin/reflect/KFunction")
         }
 
-        for (klass in listOf("Char", "Byte", "Short", "Int", "Float", "Long", "Double", "String", "Enum")) {
+        for (klass in listOf("Char", "Byte", "Short", "Int", "Float", "Long", "Double", "String", "Enum", "Boolean")) {
```
Comment (udalov) [inline core/metadata.jvm/src/org/jetbrains/kotlin/metadata/jvm/deserialization/ClassMapperLite.kt]: This change will probably confuse the compiler and reflection of older versions when they encounter a declaration with `kotlin.Boolean.Companion` in the signature. The generated bytecode will be correct, but annotations (such as `Deprecated`, for example) will not be loaded, and reflective call will throw an exception. For more information, see the comment on this class.

Please revert this change and add a short comment noting that we explicitly omit `Boolean` here. Thanks!

Comment (Kerooker) [reply core/metadata.jvm/src/org/jetbrains/kotlin/metadata/jvm/deserialization/ClassMapperLite.kt]: Done

**Comment by ilya-g** [comment]
I've fixed remaining tests and merged it into 1.3-M1 branch

**Pull request closed without merging**